lib/init.c

```diff
@@ -287,9 +289,11 @@ struct smb2_context *smb2_init_context(void)
         char buf[1024] _U_;
         int i, ret;
         static int ctr;
-
+#ifdef _XBOX
+        srandom(time(NULL) ^ ctr++);
+#else
         srandom(time(NULL) ^ getpid() ^ ctr++);
-
```

Comment:
There are already too many #ifdef _platform_ in the lib/*.c part of the library
and on later thought I think maybe better would have been to add a :
int getpid() { return 0}
to the xbox part of the compat.c/h

But It is fine, we can do those cleanups later.  I have myself also added these kinds of ifdefs which I now consider a mistake.

Reply:
I think this one is not necessary to add getpid() but it's a good idea tough i was trying this but using `#define getpid()`

Reply:
Soon i will add patch(more portable way) of getpid.

Reply:
Are you letting the xbox folks know that official support in upstream is now available and they won't need to carry an out-of-tree fork any more?

Reply:
> Are you letting the xbox folks know that official support in upstream is now available and they won't need to carry an out-of-tree fork any more?

Yes, i contacted in forums about xbox support and they stayed well about the news. 
I will send link of emuextras(which is the official place of the xbox community);
https://www.emuxtras.net/forum/viewtopic.php?f=186&t=7743
https://www.emuxtras.net/forum/viewtopic.php?f=229&t=7744

Reply:
https://github.com/sahlberg/libsmb2/pull/305

Reply:
Getpid fixes